lib/src/equality.dart

```diff
@@ -80,7 +80,7 @@ int deepHashCode(Object? value) {
   } else if (value is Iterable) {
     return const IterableEquality().hash(value.map(deepHashCode));
   } else if (value is YamlScalar) {
-    return value.value.hashCode;
+    return (value.value as Object?).hashCode;
```

Comment:
Perhaps publish `package:yaml` before we land this PR? This cast won't be necessary w/ the not-yet-published version of yaml, and will be an analysis lint (unnecessary cast).

Reply:
sure!

Reply:
is the new yaml ready to land?